openwisp_users/apps.py

```diff
@@ -1,16 +1,22 @@
 from django.apps import AppConfig
 from django.conf import settings
 from django.utils.translation import ugettext_lazy as _
+from openwisp_users import settings as app_settings
 
 
 class OpenwispUsersConfig(AppConfig):
     name = 'openwisp_users'
     app_label = 'openwisp_users'
     verbose_name = _('Users and Organizations')
 
+    rest_framework_defaults = {
+        'DEFAULT_PERMISSION_CLASSES': ['rest_framework.permissions.IsAuthenticated']
+    }
```

Comment:
yes I think we can set the bearer authentication as default here, is that what you meant? Although I'm a bit worried it may affect the existing APIs of some modules.

Reply:
BearerAuthentication class is not used so far in openwisp-users because it makes no sense to require the bearer token in order to get it from the endpoint.

BearerAuthentication class is used in openwisp-firmware-upgrader and maybe other projects. Where do we want to place this class that all projects requiring api authentication can just import it and so avoid duplicating it?

Reply:
I think openwisp-users is the right place because here is where we will deal with everything related to the management of users, organizations, multi-tenancy, permissions and I'd say API authentication is related. What do you think?

Reply:
Yes, I also think this is the correct place. Do you think we need a test for it if it's not used in this module?

Reply:
Yes we do, so that other modules don't have to write a test for it and they can assume it works (like we do with other libraries).

You can add a dummy API endpoint in the test project: https://github.com/openwisp/openwisp-users/tree/master/tests/testapp

Reply:
I have one last doubt regarding this default here, wouldn't this make any DRF view require authentication by default?
If yes, this will affect other modules which already have APIs.

Reply:
You are correct. I was not aware that other modules have already an API, so probably the best here is to add the class individually to the views that need it. Maybe we can do it in a Mixin?

Reply:
@PabloCastellano I think we can remove this from here and forget it for now. Consider this the first iteration in this module, we'll surely come back here in the near future with clearer idea of how to improve it.

At some point we'll have to add endpoints to manage the models of this module (User, Organization, Group, etc) as well, but we'll move with this gradually.